props/problem-set-timeline.checkpoint.md

```diff
@@ -0,0 +1,393 @@
+_Fig. Social media timeline web app. The top header reads "Ada Lovelace's social media feed." There are three different timeline events. Each timeline event has the timestamp "3 days ago." They also include a person's name and a status message._
+
+### `App` Component
+
+The `App` component is a container component. It is responsible for:
+
+- Holding the data about the timeline, which contains all timeline events.
+- Rendering one instance of the `Timeline` component.
+- Passing the timeline data to the `Timeline` component.
+
+Notice that our modified `App` component imports a JSON object, and declares the object in a variable named `timelineData`:
+
+```js
+import timelineData from "./data/timeline.json";
+```
+
+### `Timeline` Component
+
+The `Timeline` component is also a container component. It's responsible for:
+
+- Reading the timeline data from `App`
+- Rendering a list of `TimelineEvent` components based on the timeline data.
+- Passing person, status, and timestamp data to each `TimelineEvent` component.
+
+In other words, the `Timeline` component should receive the timeline data from `App`. For each object in the timeline data, it should render an appropriate `TimelineEvent` component.
+
+### `TimelineEvent` Component
+
+The `TimelineEvent` component is a presentational component. Its responsibilities include:
+
+- Rendering the event's status message
+- Rendering the event's author (person)
+- Rendering the event's timestamp
+
+#### The `Timestamp` Component
+
+The `TimelineEvent` component should use an instance of the `Timestamp` component.
+
+The `Timestamp` component expects one `prop`: `time`.
+
+If this component receives `time` through `props` correctly, it will render the time in a nice format!
+
+## CSS Requirement
+
+There are no requirements around CSS and styling. The CSS classes that the sample solution uses are placed in these files:
+
+- `src/App.css`
+- `src/components/Timeline.css`
+- `src/components/TimelineEvent.css`
+
+You are **_encouraged_** to use these class names in your project.
+
+## Check for Understanding
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: b050b166
+* title: Timeline
+* points: 3
+* topics: js, react
+##### !question
+
+In this assignment, which components were container components, and which components were presentational components? What are the similarities and differences of the container components and presentational components?
+
+What was the `Timestamp` component, and why?
+
+##### !end-question
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: 59d2e083
+* title: Timeline
+* points: 3
+* topics: js, react
+##### !question
+
+In your own notes, create a diagram where there are four boxes labeled:
```

Comment:
I'm not entirely clear what we're looking for here. Would the diagram be something like
```
App has an arrow to Timeline passing the events array.
Timeline has many arrows pointing to TimelineEvent, each passing a single event.
TimelineEvent has a single arrow pointing to Timestamp, passing the time of the current event.
```

Or should that last one be many arrows between TimelineEvent and Timestamp, one for each time from each event?

If either of these answers is in the vein of what you were looking for, then this prompt is probably ok, but if this isn't what you had in mind, then maybe some additional detail in the prompt would be helpful?

Reply:
I'll keep my eyes open to see if this direction feels ambiguous in the classroom!